### ISSUE TITLE
Start of the e2e test gives an error because of wrong path to type script config file.

### ISSUE THREAD

**Issue by sharikovvladislav** [ngrx/platform#241]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request
</code></pre>

## What is the current behavior?

e2e is not working.
## Expected behavior:

e2e is working :)

## Minimal reproduction of the problem with instructions:

1. clone repo
2. yarn
3. run e2e tests

Now you see this:

```
TypeError: Cannot find a tsconfig.json file at the specified directory: e2e
    at resolveSync (/Users/svlad/dev/platform/node_modules/tsconfig/src/tsconfig.ts:69:11)
    at Object.loadSync (/Users/svlad/dev/platform/node_modules/tsconfig/src/tsconfig.ts:140:16)
    at readConfig (/Users/svlad/dev/platform/node_modules/ts-node/src/index.ts:425:18)
    at Object.register (/Users/svlad/dev/platform/node_modules/ts-node/src/index.ts:189:18)
    at beforeLaunch (/Users/svlad/dev/platform/protractor.conf.js:24:24)
    at q_1.Promise (/Users/svlad/dev/platform/node_modules/protractor/lib/util.ts:48:39)
    at Function.promise (/Users/svlad/dev/platform/node_modules/q/q.js:682:9)
    at Object.runFilenameOrFn_ (/Users/svlad/dev/platform/node_modules/protractor/lib/util.ts:39:10)
    at Object.initFn [as init] (/Users/svlad/dev/platform/node_modules/protractor/lib/launcher.ts:111:10)
    at Object.<anonymous> (/Users/svlad/dev/platform/node_modules/protractor/lib/cli.ts:235:10)
```

## Version of affected browser(s),operating system(s), npm, node and ngrx:
```
sharikovvlad:platform svlad$ yarn --version
0.24.6
sharikovvlad:platform svlad$ npm -v
5.0.3
sharikovvlad:platform svlad$ sw_vers -productVersion 
10.12.5
sharikovvlad:platform svlad$ 
```

## Other information:
This can be fixed by changing `project: 'e2e'` to `project: 'e2e/tsconfig.e2e.json'`.

**Comment by brandonroberts** [comment]
Thanks for the PR 😉

**Comment by sharikovvladislav** [comment]
I can submit the PR for this. But I will open another issue right after this :)

**Comment by sharikovvladislav** [comment]
About this:

```
[01:47:07] E/protractor - Could not find Angular on page http://localhost:4200/ : retries looking for angular exceeded

Failed: Angular could not be found on the page http://localhost:4200/.If this is not an Angular application, you may need to turn off waiting for Angular.
                          Please see 
                          https://github.com/angular/protractor/blob/master/docs/timeouts.md#waiting-for-angular-on-page-load
Error: Angular could not be found on the page http://localhost:4200/.If this is not an Angular application, you may need to turn off waiting for Angular.
                          Please see 
                          https://github.com/angular/protractor/blob/master/docs/timeouts.md#waiting-for-angular-on-page-load
    at executeAsyncScript_.then (/Users/svlad/dev/platform/node_modules/protractor/lib/browser.ts:936:29)
    at ManagedPromise.invokeCallback_ (/Users/svlad/dev/platform/node_modules/selenium-webdriver/lib/promise.js:1366:14)
    at TaskQueue.execute_ (/Users/svlad/dev/platform/node_modules/selenium-webdriver/lib/promise.js:2970:14)
    at TaskQueue.executeNext_ (/Users/svlad/dev/platform/node_modules/selenium-webdriver/lib/promise.js:2953:27)
    at asyncRun (/Users/svlad/dev/platform/node_modules/selenium-webdriver/lib/promise.js:2813:27)
    at /Users/svlad/dev/platform/node_modules/selenium-webdriver/lib/promise.js:676:7
    at process._tickCallback (internal/process/next_tick.js:109:7)
From: Task: Run it("should display message saying app works") in control flow
    at Object.<anonymous> (/Users/svlad/dev/platform/node_modules/jasminewd2/index.js:94:19)
    at attemptAsync (/Users/svlad/dev/platform/node_modules/jasmine/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:3945:24)
    at QueueRunner.run (/Users/svlad/dev/platform/node_modules/jasmine/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:3884:9)
    at /Users/svlad/dev/platform/node_modules/jasmine/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:3924:18
    at /Users/svlad/dev/platform/node_modules/jasmine/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:3848:9
    at /Users/svlad/dev/platform/node_modules/jasminewd2/index.js:64:48
    at ControlFlow.emit (/Users/svlad/dev/platform/node_modules/selenium-webdriver/lib/events.js:62:21)
    at ControlFlow.shutdown_ (/Users/svlad/dev/platform/node_modules/selenium-webdriver/lib/promise.js:2565:10)
    at shutdownTask_.MicroTask (/Users/svlad/dev/platform/node_modules/selenium-webdriver/lib/promise.js:2490:53)
    at MicroTask.asyncRun (/Users/svlad/dev/platform/node_modules/selenium-webdriver/lib/promise.js:2619:9)
From asynchronous test: 
Error
    at Suite.<anonymous> (/Users/svlad/dev/platform/e2e/app.e2e-spec.ts:10:3)
    at addSpecsToSuite (/Users/svlad/dev/platform/node_modules/jasmine/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:994:25)
    at Env.describe (/Users/svlad/dev/platform/node_modules/jasmine/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:961:7)
    at describe (/Users/svlad/dev/platform/node_modules/jasmine/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:4031:18)
    at Object.<anonymous> (/Users/svlad/dev/platform/e2e/app.e2e-spec.ts:3:1)
    at Module._compile (module.js:571:32)
    at Module.m._compile (/Users/svlad/dev/platform/node_modules/ts-node/src/index.ts:392:23)
    at Module._extensions..js (module.js:580:10)
    at Object.require.extensions.(anonymous function) [as .ts] (/Users/svlad/dev/platform/node_modules/ts-node/src/index.ts:395:12)
```

**Comment by sharikovvladislav** [comment]
I did PR

**Comment by brandonroberts** [comment]
The e2e tests work for me when I run `yarn run ng -- e2e`. Will you add the following to the PR?

e2e/app-e2e.spec.ts
```ts
import { ExampleAppPage } from './app.po';

describe('example-app App', function() {
  let page: ExampleAppPage;

  beforeEach(() => {
    page = new ExampleAppPage();
  });

  it('should display the app title in the menu', () => {
    page.navigateTo();
    expect(page.getAppDescription()).toContain('Book Collection');
  });
});
```

e2e/app.po.ts
```ts
import { browser, element, by } from 'protractor';

export class ExampleAppPage {
  navigateTo() {
    return browser.get('/');
  }

  getAppDescription() {
    return element(by.css('md-toolbar-row')).getText();
  }
}
```

**Comment by sharikovvladislav** [comment]
Hm... I had been starting e2e tests by protractor plugin in WebStorm (add runner protracor, select config, etcetera). This will not work, right? May be this issue is because of I run tests wrong too?

**Comment by sharikovvladislav** [comment]
I tried to run e2e tests as you said. 

`@ngrx/store` is not available. I really don't have `@ngrx/store` in the node_modules by the way (since there is not one in the package.json file). I even tried to remove everything in the node_modules and install everything again by `yarn` command. I got this as the result:

```
sharikovvlad:platform svlad$ rm -rf node_modules/
sharikovvlad:platform svlad$ yarn  
yarn install v0.24.6
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning "ngrx-store-freeze@0.1.9" has unmet peer dependency "@ngrx/store@^2.2.1".
[4/4] 📃  Building fresh packages...
$ opencollective postinstall

     *** Thank you for using @ngrx/platform! ***

Please consider donating to our open collective
     to help us maintain this package.

  https://opencollective.com/ngrx/donate

                    ***

✨  Done in 28.59s.
sharikovvlad:platform svlad$ yarn run cli -- e2e
yarn run v0.24.6
$ "/Users/svlad/dev/platform/node_modules/.bin/ng" e2e
** NG Live Development Server is listening on localhost:49152, open your browser on http://localhost:49152 **
(node:1019) DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.      
Hash: 785001e89b6f335e2413
Time: 14387ms
chunk    {0} polyfills.bundle.js, polyfills.bundle.js.map (polyfills) 307 kB {4} [initial] [rendered]
chunk    {1} main.bundle.js, main.bundle.js.map (main) 42.8 kB {3} [initial] [rendered]
chunk    {2} styles.bundle.js, styles.bundle.js.map (styles) 45.1 kB {4} [initial] [rendered]
chunk    {3} vendor.bundle.js, vendor.bundle.js.map (vendor) 4.43 MB [initial] [rendered]
chunk    {4} inline.bundle.js, inline.bundle.js.map (inline) 0 bytes [entry] [rendered]

ERROR in /Users/svlad/dev/platform/example-app/app/core/actions/layout.ts (1,24): Cannot find module '@ngrx/store'.

ERROR in /Users/svlad/dev/platform/example-app/app/reducers/index.ts (6,8): Cannot find module '@ngrx/store'.

ERROR in /Users/svlad/dev/platform/example-app/app/auth/actions/auth.ts (1,24): Cannot find module '@ngrx/store'.

ERROR in /Users/svlad/dev/platform/example-app/app/auth/reducers/index.ts (1,55): Cannot find module '@ngrx/store'.

ERROR in /Users/svlad/dev/platform/example-app/app/core/containers/app.ts (4,23): Cannot find module '@ngrx/store'.

ERROR in /Users/svlad/dev/platform/example-app/app/auth/containers/login-page.component.ts (2,23): Cannot find module '@ngrx/store'.

ERROR in /Users/svlad/dev/platform/example-app/app/auth/services/auth-guard.service.ts (5,23): Cannot find module '@ngrx/store'.

ERROR in /Users/svlad/dev/platform/example-app/app/auth/effects/auth.effects.ts (8,33): Cannot find module '@ngrx/effects'.

ERROR in /Users/svlad/dev/platform/example-app/app/auth/auth.module.ts (6,29): Cannot find module '@ngrx/store'.

ERROR in /Users/svlad/dev/platform/example-app/app/auth/auth.module.ts (7,31): Cannot find module '@ngrx/effects'.

ERROR in /Users/svlad/dev/platform/example-app/app/app.module.ts (8,29): Cannot find module '@ngrx/store'.

ERROR in /Users/svlad/dev/platform/example-app/app/app.module.ts (9,31): Cannot find module '@ngrx/effects'.

ERROR in /Users/svlad/dev/platform/example-app/app/app.module.ts (11,45): Cannot find module '@ngrx/router-store'.

ERROR in /Users/svlad/dev/platform/example-app/app/app.module.ts (12,37): Cannot find module '@ngrx/store-devtools'.

ERROR in /Users/svlad/dev/platform/example-app/app/books/actions/book.ts (1,24): Cannot find module '@ngrx/store'.

ERROR in /Users/svlad/dev/platform/example-app/app/books/effects/book.ts (8,44): Cannot find module '@ngrx/effects'.

ERROR in /Users/svlad/dev/platform/example-app/app/books/effects/book.ts (9,24): Cannot find module '@ngrx/store'.

ERROR in /Users/svlad/dev/platform/example-app/app/books/actions/collection.ts (1,24): Cannot find module '@ngrx/store'.

ERROR in /Users/svlad/dev/platform/example-app/app/books/effects/collection.ts (7,24): Cannot find module '@ngrx/store'.

ERROR in /Users/svlad/dev/platform/example-app/app/books/effects/collection.ts (8,33): Cannot find module '@ngrx/effects'.

ERROR in /Users/svlad/dev/platform/example-app/app/books/reducers/books.ts (1,32): Cannot find module '@ngrx/store'.

ERROR in /Users/svlad/dev/platform/example-app/app/books/reducers/index.ts (1,55): Cannot find module '@ngrx/store'.

ERROR in /Users/svlad/dev/platform/example-app/app/books/guards/book-exists.ts (9,23): Cannot find module '@ngrx/store'.

ERROR in /Users/svlad/dev/platform/example-app/app/books/containers/find-book-page.ts (3,23): Cannot find module '@ngrx/store'.

ERROR in /Users/svlad/dev/platform/example-app/app/books/containers/view-book-page.ts (5,23): Cannot find module '@ngrx/store'.

ERROR in /Users/svlad/dev/platform/example-app/app/books/containers/selected-book-page.ts (2,23): Cannot find module '@ngrx/store'.

ERROR in /Users/svlad/dev/platform/example-app/app/books/containers/collection-page.ts (3,23): Cannot find module '@ngrx/store'.

ERROR in /Users/svlad/dev/platform/example-app/app/books/books.module.ts (5,29): Cannot find module '@ngrx/store'.

ERROR in /Users/svlad/dev/platform/example-app/app/books/books.module.ts (6,31): Cannot find module '@ngrx/effects'.

ERROR in Error encountered resolving symbol values statically. Could not resolve @ngrx/store relative to /Users/svlad/dev/platform/example-app/app/app.module.ts., resolving symbol AppModule in /Users/svlad/dev/platform/example-app/app/app.module.ts, resolving symbol AppModule in /Users/svlad/dev/platform/example-app/app/app.module.ts, resolving symbol AppModule in /Users/svlad/dev/platform/example-app/app/app.module.ts

ERROR in ./example-app/app/app.module.ts
Module not found: Error: Can't resolve '@ngrx/store' in '/Users/svlad/dev/platform/example-app/app'
 @ ./example-app/app/app.module.ts 13:0-42
 @ ./example-app/main.ts
 @ multi webpack-dev-server/client?http://localhost:49152 ./example-app/main.ts

ERROR in ./example-app/app/auth/auth.module.ts
Module not found: Error: Can't resolve '@ngrx/store' in '/Users/svlad/dev/platform/example-app/app/auth'
 @ ./example-app/app/auth/auth.module.ts 12:0-42
 @ ./example-app/app/app.module.ts
 @ ./example-app/main.ts
 @ multi webpack-dev-server/client?http://localhost:49152 ./example-app/main.ts

ERROR in ./example-app/app/reducers/index.ts
Module not found: Error: Can't resolve '@ngrx/store' in '/Users/svlad/dev/platform/example-app/app/reducers'
 @ ./example-app/app/reducers/index.ts 1:0-69
 @ ./example-app/app/app.module.ts
 @ ./example-app/main.ts
 @ multi webpack-dev-server/client?http://localhost:49152 ./example-app/main.ts

ERROR in ./example-app/app/core/containers/app.ts
Module not found: Error: Can't resolve '@ngrx/store' in '/Users/svlad/dev/platform/example-app/app/core/containers'
 @ ./example-app/app/core/containers/app.ts 12:0-36
 @ ./example-app/app/app.module.ts
 @ ./example-app/main.ts
 @ multi webpack-dev-server/client?http://localhost:49152 ./example-app/main.ts

ERROR in ./example-app/app/app.module.ts
Module not found: Error: Can't resolve '@ngrx/effects' in '/Users/svlad/dev/platform/example-app/app'
 @ ./example-app/app/app.module.ts 14:0-46
 @ ./example-app/main.ts
 @ multi webpack-dev-server/client?http://localhost:49152 ./example-app/main.ts

ERROR in ./example-app/app/auth/auth.module.ts
Module not found: Error: Can't resolve '@ngrx/effects' in '/Users/svlad/dev/platform/example-app/app/auth'
 @ ./example-app/app/auth/auth.module.ts 13:0-46
 @ ./example-app/app/app.module.ts
 @ ./example-app/main.ts
 @ multi webpack-dev-server/client?http://localhost:49152 ./example-app/main.ts

ERROR in ./example-app/app/app.module.ts
Module not found: Error: Can't resolve '@ngrx/router-store' in '/Users/svlad/dev/platform/example-app/app'
 @ ./example-app/app/app.module.ts 16:0-65
 @ ./example-app/main.ts
 @ multi webpack-dev-server/client?http://localhost:49152 ./example-app/main.ts

ERROR in ./example-app/app/app.module.ts
Module not found: Error: Can't resolve '@ngrx/store-devtools' in '/Users/svlad/dev/platform/example-app/app'
 @ ./example-app/app/app.module.ts 17:0-59
 @ ./example-app/main.ts
 @ multi webpack-dev-server/client?http://localhost:49152 ./example-app/main.ts

ERROR in ./example-app/app/auth/services/auth-guard.service.ts
Module not found: Error: Can't resolve '@ngrx/store' in '/Users/svlad/dev/platform/example-app/app/auth/services'
 @ ./example-app/app/auth/services/auth-guard.service.ts 13:0-36
 @ ./example-app/app/routes.ts
 @ ./example-app/app/app.module.ts
 @ ./example-app/main.ts
 @ multi webpack-dev-server/client?http://localhost:49152 ./example-app/main.ts

ERROR in ./example-app/app/auth/containers/login-page.component.ts
Module not found: Error: Can't resolve '@ngrx/store' in '/Users/svlad/dev/platform/example-app/app/auth/containers'
 @ ./example-app/app/auth/containers/login-page.component.ts 11:0-36
 @ ./example-app/app/auth/auth.module.ts
 @ ./example-app/app/app.module.ts
 @ ./example-app/main.ts
 @ multi webpack-dev-server/client?http://localhost:49152 ./example-app/main.ts

ERROR in ./example-app/app/auth/effects/auth.effects.ts
Module not found: Error: Can't resolve '@ngrx/effects' in '/Users/svlad/dev/platform/example-app/app/auth/effects'
 @ ./example-app/app/auth/effects/auth.effects.ts 17:0-48
 @ ./example-app/app/auth/auth.module.ts
 @ ./example-app/app/app.module.ts
 @ ./example-app/main.ts
 @ multi webpack-dev-server/client?http://localhost:49152 ./example-app/main.ts

ERROR in ./example-app/app/auth/reducers/index.ts
Module not found: Error: Can't resolve '@ngrx/store' in '/Users/svlad/dev/platform/example-app/app/auth/reducers'
 @ ./example-app/app/auth/reducers/index.ts 1:0-68
 @ ./example-app/app/auth/auth.module.ts
 @ ./example-app/app/app.module.ts
 @ ./example-app/main.ts
 @ multi webpack-dev-server/client?http://localhost:49152 ./example-app/main.ts
webpack: Failed to compile.
[11:01:13] I/update - chromedriver: file exists /Users/svlad/dev/platform/node_modules/webdriver-manager/selenium/chromedriver_2.31.zip
[11:01:13] I/update - chromedriver: unzipping chromedriver_2.31.zip
[11:01:14] I/update - chromedriver: setting permissions to 0755 for /Users/svlad/dev/platform/node_modules/webdriver-manager/selenium/chromedriver_2.31
[11:01:14] I/update - chromedriver: chromedriver_2.31 up to date
[11:01:14] I/launcher - Running 1 instances of WebDriver
[11:01:14] I/direct - Using ChromeDriver directly...
Spec started
[11:01:25] E/protractor - Could not find Angular on page http://localhost:49152/ : retries looking for angular exceeded

  example-app App
    ✗ should display the app title in the menu
      - Failed: Angular could not be found on the page http://localhost:49152/.If this is not an Angular application, you may need to turn off waiting for Angular.
                                Please see 
                                https://github.com/angular/protractor/blob/master/docs/timeouts.md#waiting-for-angular-on-page-load
                                Please see 
                                https://github.com/angular/protractor/blob/master/docs/timeouts.md#waiting-for-angular-on-page-load
          at executeAsyncScript_.then (/Users/svlad/dev/platform/node_modules/protractor/lib/browser.ts:936:29)
          at ManagedPromise.invokeCallback_ (/Users/svlad/dev/platform/node_modules/selenium-webdriver/lib/promise.js:1366:14)
          at TaskQueue.execute_ (/Users/svlad/dev/platform/node_modules/selenium-webdriver/lib/promise.js:2970:14)
          at TaskQueue.executeNext_ (/Users/svlad/dev/platform/node_modules/selenium-webdriver/lib/promise.js:2953:27)
          at asyncRun (/Users/svlad/dev/platform/node_modules/selenium-webdriver/lib/promise.js:2813:27)
          at /Users/svlad/dev/platform/node_modules/selenium-webdriver/lib/promise.js:676:7
          at process._tickCallback (internal/process/next_tick.js:109:7)
      From: Task: Run it("should display the app title in the menu") in control flow
          at Object.<anonymous> (/Users/svlad/dev/platform/node_modules/jasminewd2/index.js:94:19)
          at /Users/svlad/dev/platform/node_modules/jasminewd2/index.js:64:48
          at ControlFlow.emit (/Users/svlad/dev/platform/node_modules/selenium-webdriver/lib/events.js:62:21)
          at ControlFlow.shutdown_ (/Users/svlad/dev/platform/node_modules/selenium-webdriver/lib/promise.js:2565:10)
          at shutdownTask_.MicroTask (/Users/svlad/dev/platform/node_modules/selenium-webdriver/lib/promise.js:2490:53)
          at MicroTask.asyncRun (/Users/svlad/dev/platform/node_modules/selenium-webdriver/lib/promise.js:2619:9)
          at /Users/svlad/dev/platform/node_modules/selenium-webdriver/lib/promise.js:676:7
          at process._tickCallback (internal/process/next_tick.js:109:7)
      From asynchronous test: 
      Error
          at Suite.<anonymous> (/Users/svlad/dev/platform/e2e/app.e2e-spec.ts:10:3)
          at Object.<anonymous> (/Users/svlad/dev/platform/e2e/app.e2e-spec.ts:3:1)
          at Module._compile (module.js:571:32)
          at Module.m._compile (/Users/svlad/dev/platform/node_modules/ts-node/src/index.ts:392:23)
          at Module._extensions..js (module.js:580:10)
          at Object.require.extensions.(anonymous function) [as .ts] (/Users/svlad/dev/platform/node_modules/ts-node/src/index.ts:395:12)
          at Module.load (module.js:488:32)
          at tryModuleLoad (module.js:447:12)
          at Function.Module._load (module.js:439:3)
          at Module.require (module.js:498:17)
          at require (internal/module.js:20:19)

**************************************************
*                    Failures                    *
**************************************************

1) example-app App should display the app title in the menu
  - Failed: Angular could not be found on the page http://localhost:49152/.If this is not an Angular application, you may need to turn off waiting for Angular.
                            Please see 
                            https://github.com/angular/protractor/blob/master/docs/timeouts.md#waiting-for-angular-on-page-load

Executed 1 of 1 spec (1 FAILED) in 10 secs.
[11:01:25] I/launcher - 0 instance(s) of WebDriver still running
[11:01:25] I/launcher - chrome #01 failed 1 test(s)
[11:01:25] I/launcher - overall: 1 failed spec(s)
[11:01:25] E/launcher - Process exited with error code 1
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
sharikovvlad:platform svlad$ 
```

By the way, this page https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md says that I should use `npm` as a package manager. Is it neccessary. What is ngrx package manager? May be contributing page should recommend yarn way?

**Comment by sharikovvladislav** [comment]
I don't see same when I run through webstorm but I think it is because ws supress all of these warnings.

**Comment by sharikovvladislav** [comment]
@brandonroberts any thought on this?

**Comment by sharikovvladislav** [comment]
@brandonroberts total magic. 

I don't know how but now everything is working...

What I did:
1) I was seeing on error message I mentioned there: https://github.com/ngrx/platform/issues/241#issuecomment-320598120
2) I tried to start example app with `yarn run example:start` to see if example-app is working even without node_modules in the `example-app` folder
3) I tried to run `yarn run cli -- e2e`.

And it worked.... Omg... total magic.

I think this should be closed now. Lets discuss the code now inside the pr.

PS: by the way - I wanted to fix tsconfig path to fix e2e test so they will work out the box. They didn't work since ngrx/example-app repo :( Ty for your code there: https://github.com/ngrx/platform/issues/241#issuecomment-320542674
